pkg/test/expectations/expectations.go

```diff
@@ -27,3 +29,24 @@ import (
 func ExpectUnavailable(env *test.Environment, instanceType string, zone string, capacityType string) {
 	Expect(env.UnavailableOfferingsCache.IsUnavailable(instanceType, fmt.Sprintf("%s-%s", fake.Region, zone), capacityType)).To(BeTrue())
 }
+
+func ExpectKubeletFlags(env *test.Environment, customData string, expectedFlags map[string]string) {
+	kubeletFlags := customData[strings.Index(customData, "KUBELET_FLAGS=")+len("KUBELET_FLAGS="):]
+	for flag, value := range expectedFlags {
+		Expect(kubeletFlags).To(ContainSubstring(fmt.Sprintf("--%s=%s", flag, value)))
+	}
+}
+
+func AssertDecodeCustomData(env *test.Environment) string {
+	Expect(env.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+
+	vm := env.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+	customData := *vm.Properties.OSProfile.CustomData
+	Expect(customData).ToNot(BeNil())
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(customData)
+	Expect(err).To(Succeed())
+	decodedString := string(decodedBytes[:])
+
+	return decodedString
+}
```

Comment:
Do we think its worth creating a separate package for assertions? Is assertDecodeCustomData a good name?

Reply:
Maybe `ExpectDecodeVMCustomData`? `Expect` is the most commonly used within our testing, and VM in case there is any another source of custom data?

Reply:
The reason I didnt do Expect was it was sort of returning a value, but Expect does make more sense with the consistent patterns we have throughout the codebase. We can set it to expect